### PR TITLE
add autoflake to remove unused imports

### PR DESCRIPTION
**Pull request by XuesongYang**
Signed-off-by: Xuesong Yang <1646669+XuesongYang@users.noreply.github.com>

# What does this PR do ?
This hook does following process. I also manually add `# noqa` to avoid changes for some lines of codes for safety purpose. Domain experts please help to review. 
```
# after process, we fixed,
22 for Unused import
1 for 'import *' may pollute namespace
```
```
  --expand-star-imports
                        expand wildcard star imports with undefined names; this only triggers if there is only one star import in the file; this is skipped if there are any uses of `__all__` or `del` in the file
  --remove-all-unused-imports
                        remove all unused imports (not just those from the standard library)
  --ignore-init-module-imports
                        exclude __init__.py when removing unused imports
  --ignore-pass-after-docstring
                        keep `pass` statements after a new line ending on """
```

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/data/language_modeling/megatron/dataset_utils.py
Comment (MaximumEntropy) [inline nemo/collections/nlp/data/language_modeling/megatron/dataset_utils.py:1077]: Please revert these. These need to be imported here to avoid cyclic imports.

**Review by MaximumEntropy — CHANGES_REQUESTED** [review]
Thanks for cleaning stuff up! Some of the imports are still relevant though, to avoid cyclic imports.

File: nemo/collections/nlp/data/language_modeling/megatron/dataset_utils.py
Comment (XuesongYang) [reply nemo/collections/nlp/data/language_modeling/megatron/dataset_utils.py:1077]: Thanks for your comments. Fixed in next revision.

Comment (MaximumEntropy) [reply nemo/collections/nlp/data/language_modeling/megatron/dataset_utils.py:1077]: Thank you!

**Comment by MaximumEntropy** [comment]
Looks good from my side, will leave it to other code-owners to approve.

**Comment by titu1994** [comment]
I too am worried about cyclic imports + backward compat import paths. Question is does this solve something or is just a nice thing to have for the sake of cleanliness. 

If it must be merged, hold off until we cut the r1.13 branch, cause I don't want a release to have breaking changes due to some style fixes.

**Comment by titu1994** [comment]
Imo there is no "danger" or "extreme downside" of having excess imports - on the flip side having to work against a auto tool with these #noqa flags hinders development in the sake of "fixing" dangling imports (many of which are needed simply for hydra instantiation or even just convenience). 

My take is it's a pedantic approach to clean up imports and can cause more harm than benefit, and requires more work to fight against a barebones rule that doesn't really hold much value

**Review by titu1994 — CHANGES_REQUESTED** [review]
I'd rather not have to fight against a commit prehook that silently changes my imports.

**Comment by XuesongYang** [comment]
> I'd rather not have to fight against a commit prehook that silently changes my imports.

That makes sense. Let's close this PR to avoid potential bugs. Thank you guys for the comments!

**Pull request closed without merging**
